sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_pipeline_expression.py

```diff
@@ -456,7 +465,7 @@ def _validate_operand_type(_type: str):
     def _get_operand_type(self, operand: str) -> str:
         if operand in self._inputs:
             return self._inputs[operand].type
-        return type(eval(operand)).__name__  # pylint: disable=eval-used # nosec
+        return IOConstants.PRIMITIVE_TYPE_2_STR[type(eval(operand))]  # pylint: disable=eval-used # nosec
```

Comment:
is that possible operand being None?

Reply:
no, `None` is currently not supported in expression (`None` is meaningless for numeric, comparison and bitwise logical operators), so it shouldn't appear here.

Reply:
Also _validate_operand_type above, how about None?

Reply:
how do we support boolean parameter comparison?

Reply:
from my point of view, I don't want to support operation `param == None`, it should be clear `True` or `False`; besides, comparing to `None` might better use `is` rather than `==` ([PEP 8 - Programming Recommendations](https://peps.python.org/pep-0008/#programming-recommendations)). However, if we do have case that uses `==` and `None`, we can make it valid.
BTW, `None` is not supported in expression in v1.5, so it might be OK to ban `None` in expression for now?

Reply:
offline synced on this, shall support `None` in expression

Reply:
support `None` in new commit 7ec2af6.